Initialize achievements and the `AchievementManager` on the server

drunderscore opened the pull request:
This formally fixes #2661

See the first commit's message for fine details about the fix itself.

Note that these changes should **never** cause achievements to be awarded or saved on the server. Some methods in `AchievementsHelper` check the `netMode` to ensure it is not on the server. For the methods that do not, it is either checked at the call site that it is not the server, or simply unreachable by control flow on the server. If you do know of an instance where the server could award an achievement, do let me know, because **it is undesirable**.

Edit: To visualize what this PR is actually achieving (get it??), here's a demonstration

Chat message:
```
woah i just got the [a:KILL_THE_SUN] achievement!
thats way better than the [a:TOPPED_OFF] achievement
neither of those compare to the [a:THIS_IS_NOT_VALID_LMAO] achievement
```

Console (no longer throws, and properly logs message):
![image](https://user-images.githubusercontent.com/15949431/205384928-e8c717dd-cfb0-4afc-a1b4-a31dfd3a78f6.png)

drunderscore committed:
Initialize achievements and the `AchievementManager` on the server

The `AchievementTagHandler` expects `Main.Achievements` to be non-null,
which is not normally the case on dedicated servers. When trying to
parse an achievement chat tag, it will instead throw.

The tag is parsed when calling `ChatManager.ParseMessage`, which is used
in TShock when writing chat messages to the console. Our `OnChat`
handler uses `Utils.Broadcast`, which will send the message to all
connected clients, write the message to the console and the log. Due to
the order of execution, the message ends up being sent to all connected
clients, but throws whilst trying to write to the console, and never
gets written to the log.

To solve the issue, we make achievements available on the server,
allowing the tag handler to work as expected, and even allowing the
localization of achievements names to appear in the console.

drunderscore committed:
Update `docs/changelog.md`

punchready commented:
How would the server even (accidentally) award achievements? There is no packet for it, only for NPC kills in packet 97, invasions in packet 98, and a few implicit ones from received packets, all of which are obviously part of the normal functionality.

drunderscore commented:
@punchready By award achievements on the server, I meant the server invoking `AchivementHelper` methods and awarding them. Like I said it shouldn't happen, and it would already be a problem now if it was happening.

drunderscore committed:
Allow multiple test cases in the TShock test suite

Previously, we were initializing TShock before each test. This isn't
actually what we want to do, because NUnit will (by default) execute all
tests within one instance. Initializing TShock multiple times would
cause issues, so let's just do it once at the very beginning, before any
tests execute.

drunderscore committed:
Add tests for parsing achievement tags in chat

These tests ensure that `ChatManager.ParseMessage` will not throw on the
server with TShock when given an achievement tag.

drunderscore commented:
Alright, this is now ready for review. Apologies about the somewhat-unrelated test suite changes in 0ac814c, but I wanted to add a test for this and was unable to without these changes.

CC @SignatureBeef for the changes to the test suite, and the test itself making sense.

hakusaro commented:
This is good-to-go if @SignatureBeef is okay with it.

SignatureBeef reviewed (approved):
all seems reasonable to me 👍

hakusaro commented:
Merged as of 24bb002e646c88dacb1d1d864f8c171b48784435 (I fucked the rebase up 🙃 )
